src/main/java/com/example/b2b_opportunities/Service/CompanyService.java

```diff
@@ -176,10 +180,30 @@ public void deleteCompanyImage(Authentication authentication) {
         delete(authentication, "image");
     }
 
-    public List<ProjectResponseDto> getCompanyProjects(Long companyId) {
+    public Set<ProjectResponseDto> getCompanyProjects(Authentication authentication, Long companyId) {
         Company company = companyRepository.findById(companyId)
                 .orElseThrow(() -> new NotFoundException("Company with ID: " + companyId + " not found"));
-        return ProjectMapper.toDtoList(company.getProjects());
+        Company userCompany = getUserCompanyOrThrow(userService.getCurrentUserOrThrow(authentication));
+        if (userCompany.getId().equals(companyId)) {
+            //if logged user is checking his projects - return all of them
+            return ProjectMapper.toDtoSet(new HashSet<>(company.getProjects()));
+        }
+        if (!company.isApproved()) {
+            throw new NotFoundException("This company has not yet been approved to post public projects");
```

Comment:
this should return empty list instead. No need for errors. This can be logged instead (once we have logging)